File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        int compare = 0;
        int number = x;
        while (number > 0) {
            compare = compare * 10 + number % 10;
            number = number / 10;
        }
        if (compare == x) {
            return true;
        }
        return false;
    }
}

File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0) return false;

        long long comp = 0, num = x;

        while (num > 0) {
            comp = comp * 10 + num % 10;
            num /= 10;
        }

        return comp == x;
    }
};